edge-modules/MetricsCollector/MetricsScrapeAndUpload.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace MetricsCollector
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Agent.Diagnostics;
+    using Microsoft.Azure.Devices.Edge.Agent.Diagnostics.Publisher;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class MetricsScrapeAndUpload : IDisposable
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("MetricsCollector");
+        readonly IMetricsScraper scraper;
+        readonly IMetricsPublisher publisher;
+        PeriodicTask periodicScrapeAndUpload;
+        Guid testId; // used for differentiating between test runs on the same device
+
+        public MetricsScrapeAndUpload(IMetricsScraper scraper, IMetricsPublisher publisher, Guid testId)
```

Comment:
why call testId? this unique information can only be used for testing?

Reply:
Good point, as others might use this module we should call it something like metricsCollectorRuntimeId